### PR TITLE
Clarify ICE consent freshness feedback

### PR DESCRIPTION
**Pull request by aboba**
Fix for Issue https://github.com/w3c/webrtc-pc/issues/517

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -4415,8 +4415,9 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
 
           <dd>The <code><a>RTCIceTransport</a></code> has received at least one
           remote candidate, and is checking candidate pairs but has not yet
-          found a connection. In addition to checking, it may also still be
-          gathering.</dd>
+          found a connection, or consent checks [[!RFC7675]] have failed on a
```
Comment (taylor-b) [inline webrtc.html]: Not a big deal, but it's not 100% clear that the boolean logic is "A and (B or C)" and not "(A and B) or C". Maybe adding "either" would help: "... has received at least remote candidate, and either is checking ..."

File: webrtc.html [hunk 2]
```diff
@@ -4429,7 +4430,9 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
 
           <dd>The <code><a>RTCIceTransport</a></code> has finished gathering,
           received an indication that there are no more remote candidates,
-          finished checking all candidate pairs and found a connection.</dd>
+          finished checking all candidate pairs and found a connection.
+          If consent checks [[!RFC7675]] subsequently fail, the state 
+          transitions to "failed".</dd>
 
           <dt>failed</dt>
```
Comment (taylor-b) [inline webrtc.html]: Don't you need to add something in the "failed" section"?

File: webrtc.html [hunk 3]
```diff
@@ -4429,7 +4430,9 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
 
           <dd>The <code><a>RTCIceTransport</a></code> has finished gathering,
           received an indication that there are no more remote candidates,
-          finished checking all candidate pairs and found a connection.</dd>
+          finished checking all candidate pairs and found a connection.
+          If consent checks [[!RFC7675]] subsequently fail, the state
```
Comment (taylor-b) [inline webrtc.html]: If we're going to add this sentence in "completed", should we do the same in "connected"? "If consent checks fail, the state transitions to disconnected"?

**Comment by taylor-b** [comment]
Besides the comments, I have a broader question about this change. If the candidate pair currently being used loses consent, but another candidate pair still has consent, can the IceTransport switch to it? I believe that's what Chrome would currently do. In that case, the state should only be "failed" if _every_ pair either failed checking or lost consent.

**Comment by aboba** [comment]
@taylor-b Agree that state should only be "failed" if every pair either failed checking or lost consent. Also, if state was "connected" and consent is lost on the candidate pair currently being used, but another candidate pair still has consent, state is still "connected", right? (e.g. no transition to "disconnected").

File: webrtc.html (hunk 3 above)
Comment (aboba) [inline webrtc.html]: If consent checks fail in "connected" the state transitions to "disconnected" only if there are no successful candidate pairs left.

File: webrtc.html [hunk 4]
```diff
@@ -4414,37 +4414,43 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
           <dt>checking</dt>
 
           <dd>The <code><a>RTCIceTransport</a></code> has received at least one
-          remote candidate, and is checking candidate pairs but has not yet
-          found a connection. In addition to checking, it may also still be
-          gathering.</dd>
+          remote candidate, and either is checking candidate pairs but has not yet
```
Comment (taylor-b) [inline webrtc.html]: It looks like I suggested putting the "either" in the wrong spot; my apologies. The intended logic is "received candidate AND is checking AND (no connection found OR consent checks failed)"

File: webrtc.html [hunk 5]
```diff
@@ -4414,37 +4414,43 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
...
 
           <dt>connected</dt>
 
           <dd>The <code><a>RTCIceTransport</a></code> has found a usable
           connection, but is still checking other candidate pairs to see if
           there is a better connection. It may also still be gathering and/or
-          waiting for additional remote candidates.</dd>
+          waiting for additional remote candidates. If consent checks [[!RFC7675]]
+          fail on the connection in use, and there are no other successful
+          candidate pairs available, then the state transitions to "disconnected".</dd>
```
Comment (taylor-b) [inline webrtc.html]: I guess it could transition to "checking" as well, depending on whether or not the transport is currently checking.

**Comment by taylor-b** [comment]
@aboba I agree that if a pair loses consent but the IceTransport can seamlessly switch to a different pair, the state should stay "connected". The only necessary indication for this is a `selectedcandidatepairchange` event.

File: webrtc.html (hunk 3 above)
Comment (aboba) [inline webrtc.html]: If consent checks fail in "connected" and there are no successful candidate pairs left, the state transitions to "disconnected" if there are no candidate pairs left to check (but gathering is not complete or the last remote candidate has not been received), or to "checking" if there are candidate pairs left to check.

File: webrtc.html [hunk 6]
```diff
@@ -4484,8 +4493,8 @@ <h2 id="sec.cert-mgmt">Certificate Management</h2>
           <li>(<code>connected</code>, finished all checks):
           <code>completed</code></li>
 
-          <li>(<code>completed</code>, lost connectivity):
-          <code>disconnected</code></li>
```
Comment (taylor-b) [inline webrtc.html]: The "lost connectivity" transition to "disconnected" is still possible, so I don't think this needs to be removed.

**Comment by taylor-b** [comment]
lgtm with a minor comment (take it or leave it)

**Comment by nils-ohlmeier** [comment]
lgtm

**Comment by elagerway** [comment]
lgtm

**Pull request merged**